tests/test_misc.py

```diff
@@ -10,47 +15,41 @@
-        self.assertTrue(value[0].startswith('add'))
+def test_apply():
+    oopb = _cmn.onnx_ops.OnnxOperatorBuilder(_cmn.OnnxObjectContainer(get_maximum_opset_supported()),
+                                             _cmn.InterimContext('_curr'))
+    value = oopb.apply_add((np.array([[1.0], [0.5]], dtype='float32'),
+                            ('_i1', oopb.float, np.array([2.0], dtype='float32'))), 'add')
+    assert value[0].startswith('add')
 
-    @unittest.skipIf(StrictVersion(onnx.__version__) < StrictVersion("1.2"),
-                     "Not supported in ONNX version less than 1.2, since this test requires opset 7.")
-    def test_model_creation(self):
-        N, C, H, W = 2, 3, 5, 5
-        input1 = keras.layers.Input(shape=(H, W, C))
-        x1 = keras.layers.Dense(8, activation='relu')(input1)
-        input2 = keras.layers.Input(shape=(H, W, C))
-        x2 = keras.layers.Dense(8, activation='relu')(input2)
-        maximum_layer = keras.layers.Maximum()([x1, x2])
 
-        out = keras.layers.Dense(8)(maximum_layer)
-        model = keras.models.Model(inputs=[input1, input2], outputs=out)
+@pytest.mark.skipif(StrictVersion(onnx.__version__) < StrictVersion("1.2"),
+                 reason="Not supported in ONNX version less than 1.2, since this test requires opset 7.")
+def test_model_creation():
+    N, C, H, W = 2, 3, 5, 5
+    input1 = keras.layers.Input(shape=(H, W, C))
+    x1 = keras.layers.Dense(8, activation='relu')(input1)
+    input2 = keras.layers.Input(shape=(H, W, C))
+    x2 = keras.layers.Dense(8, activation='relu')(input2)
+    maximum_layer = keras.layers.Maximum()([x1, x2])
 
-        trial1 = np.random.rand(N, H, W, C).astype(np.float32, copy=False)
-        trial2 = np.random.rand(N, H, W, C).astype(np.float32, copy=False)
+    out = keras.layers.Dense(8)(maximum_layer)
+    model = keras.models.Model(inputs=[input1, input2], outputs=out)
 
-        predicted = model.predict([trial1, trial2])
-        self.assertIsNotNone(predicted)
+    trial1 = np.random.rand(N, H, W, C).astype(np.float32, copy=False)
+    trial2 = np.random.rand(N, H, W, C).astype(np.float32, copy=False)
 
-        converted_model_7 = keras2onnx.convert_keras(model, target_opset=7)
-        converted_model_5 = keras2onnx.convert_keras(model, target_opset=5)
+    predicted = model.predict([trial1, trial2])
+    assert predicted is not None
 
-        self.assertIsNotNone(converted_model_7)
-        self.assertIsNotNone(converted_model_5)
+    converted_model_7 = keras2onnx.convert_keras(model, target_opset=7)
+    converted_model_5 = keras2onnx.convert_keras(model, target_opset=5)
 
-        opset_comparison = converted_model_7.opset_import[0].version > converted_model_5.opset_import[0].version
+    assert converted_model_7 is not None
+    assert converted_model_5 is not None
 
-        self.assertTrue(opset_comparison)
-
-
-if __name__ == '__main__':
```

Comment:
we sometimes run this single test file with python cmd line instead of pytest, which helps on debugging. Is there alternative way in the pytest as well?

Reply:
You can select a specific file by running `pytest` against that file.
```
pytest tests/test_misc.py -vv
```
You can also filter tests for specific strings of interest (e.g. LSTM):
```
pytest tests -k LSTM -vv
```